src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessorUnix.cs

```diff
@@ -127,6 +136,74 @@ public override void Delete(string mapName, MemoryMappedFile mmf)
             }
         }
 
+        /// <summary>
+        /// Checks if a list of directories is specified in AppSettings to create <see cref="MemoryMappedFile"/>.
+        /// If one is specified, returns that list. Otherwise returns the default list.
+        /// </summary>
+        /// <returns>List of paths of directories where <see cref="MemoryMappedFile"/> are allowed to be created.</returns>
+        internal List<string> GetAllowedDirectories()
+        {
+            string envVal = _environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionsUnixSharedMemoryDirectories);
+            if (string.IsNullOrEmpty(envVal))
+            {
+                return SharedMemoryConstants.TempDirs;
+            }
+
+            return envVal.Split(',').ToList();
+        }
+
+        /// <summary>
+        /// From a list of allowed directories where <see cref="MemoryMappedFile"/> can be created, it will return
+        /// a list of those that are valid (i.e. exist, or have been successfully created).
+        /// </summary>
+        /// <returns>List of paths of directories where <see cref="MemoryMappedFile"/> can be created.</returns>
+        internal List<string> GetValidDirectories()
+        {
+            List<string> allowedDirectories = GetAllowedDirectories();
+            List<string> validDirectories = new List<string>();
+
+            foreach (string directory in allowedDirectories)
+            {
+                string path = Path.Combine(directory, SharedMemoryConstants.TempDirSuffix);
+                if (Directory.Exists(path))
+                {
+                    Logger.LogTrace("Found directory for shared memory usage: {Directory}", path);
+                    try
+                    {
+                        // If the directory already exists (maybe from a previous run of the host) then clean it up and start afresh
+                        // The previously created memory maps in that directory are not needed and we need to should clean up the memory
+                        Directory.Delete(path, recursive: true);
+                        Logger.LogTrace("Cleaned up existing directory for shared memory usage: {Directory}", path);
+                    }
+                    catch (Exception exception)
+                    {
+                        Logger.LogWarning(exception, "Cannot delete existing directory for shared memory usage: {Directory}", path);
+                    }
+                }
+
+                try
+                {
+                    DirectoryInfo info = Directory.CreateDirectory(path);
+                    if (info.Exists)
+                    {
+                        validDirectories.Add(path);
+                        Logger.LogTrace("Created directory for shared memory usage: {Directory}", path);
+                    }
+                    else
+                    {
+                        Logger.LogWarning("Cannot create directory for shared memory usage: {Directory}", path);
```

Comment:
be more explicit in logging to distinguish failure vs no exists- `Directory for shared memory usage: {Directory}, does not exist`